src/NSwag.Generation.AspNetCore/AspNetCoreOpenApiDocumentGenerator.cs

```diff
@@ -440,19 +446,23 @@ private string GetOperationId(OpenApiDocument document, ControllerActionDescript
             {
                 operationId = actionDescriptor.AttributeRouteInfo.Name;
             }
-            else
+            else if (actionDescriptor is ControllerActionDescriptor controllerActionDescriptor)
             {
-                dynamic openApiControllerAttribute = actionDescriptor
+                dynamic openApiControllerAttribute = controllerActionDescriptor
                     .ControllerTypeInfo?
                     .GetCustomAttributes()?
                     .FirstAssignableToTypeNameOrDefault("OpenApiControllerAttribute", TypeNameStyle.Name);
 
                 var controllerName =
                     openApiControllerAttribute != null && !string.IsNullOrEmpty(openApiControllerAttribute.Name) ?
                     openApiControllerAttribute.Name :
-                    actionDescriptor.ControllerName;
+                    controllerActionDescriptor.ControllerName;
 
-                operationId = controllerName + "_" + GetActionName(actionDescriptor.ActionName);
+                operationId = controllerName + "_" + GetActionName(controllerActionDescriptor.ActionName);
+            }
+            else
+            {
+                operationId = Guid.NewGuid().ToString(); // TODO: What operation ID to use for minimal APIs?
```

Comment:
Here we need to construct an operation ID based on the HTTP method and path/route.

Reply:
Fixed.